rmf_websocket/test/test_client_with_update_cb.cpp

```diff
@@ -96,35 +109,35 @@ TEST_CASE("Client", "Reconnecting server") {
   nlohmann::json jsonString;
   jsonString["test"] = "1";
   broadcaster->publish(jsonString);
+  REQUIRE_NOTHROW(on_init_promise.get_future().wait_for(5s));
+
   t1.join();
 
+  REQUIRE_NOTHROW(on_server_down.get_future().wait_for(5s));
+  on_server_down = std::promise<void>();
```

Comment:
This promise shouldn't be need, `t1.join()` should implicitly mean that the server has closed.